OCD-4107: ICS Errors Report table updates

kekey1 opened the pull request.

kekey1 committed:
feat: Add certified_product_id column to ICS errors report table

This will replace the 'url' column.

[#OCD-4107]

kekey1 committed:
Merge branch 'staging' into OCD-4107

kekey1 committed:
feat: Allow 'url' column of ICS errors table to be null

[#OCD-4107]

kekey1 committed:
Merge branch 'staging' into OCD-4107

andlar commented on changes/ocd-4107.sql:
```diff
@@ -0,0 +1,26 @@
+
+-- Allow URL field to be null and we won't fill it in any longer
+ALTER TABLE openchpl.inheritance_errors_report
+ALTER COLUMN url DROP NOT NULL;
```
What _was_ this column used for? Is there some functionality that's going away?

kekey1 replied:
We were storing the URL to the listing in the database and putting it into the report. I changed it so that the database stores the certified_product_id and the report emailing job builds the URL with that ID + a property we already have. There are several other columns I considered deleting and just deriving in the report emailing job but did not. Can you try to remember to ask about this either at the FR or maybe we could just have a developer chat to see if you guys think it's important to remove  any others?

kekey1 committed:
Merge branch 'staging' into OCD-4107

kekey1 committed:
feat: Drop many ICS error columns that can be derived from a listing

Going forward, we plan to treat all "reporting" tables in this way. We
will do a regular delete of the data and insert all of the new data at
once. There is no need to keep derivable columns in the reporting tables
if they can be gotten efficiently from listings via the shared store.

[#OCD-4107]

kekey1 committed:
Merge branch 'staging' into OCD-4107

kekey1 committed:
Merge branch 'staging' into OCD-4107
